subsystems/visionsubsystem.py

```diff
@@ -72,6 +87,31 @@ def periodic(self) -> None:
         robotPose = SmartDashboard.getNumberArray(
             constants.kRobotPoseArrayKeys.valueKey, [0, 0, 0]
         )
+
+        noteResult = self.noteCamera.getLatestResult()
+        if noteResult.hasTargets():
+            notes = noteResult.getTargets()
+            notePositions = [
+                Pose3d(robotPose[0], robotPose[1], 0, Rotation3d(0, 0, robotPose[2]))
+                + constants.kRobotToNoteCameraTransform
+                + self.getCameraToNote(note)
+                for note in notes
+            ]
+            closestNote = Pose2d(*robotPose).nearest(notePositions)
+            intakePickupPosition = robotPose + constants.kRobotToIntakePickupTransform
+
+            # angle robot needs to rotate by to pick up note by driving forward
+            self.dRobotAngle = (
+                Rotation2d(robotPose[2])
+                - Transform2d(intakePickupPosition, closestNote).rotation()
+            )
+
+            SmartDashboard.putBoolean(constants.kNoteInViewKey, True)
+        else:
+            # rotate around if no note in vision
+            self.dRobotAngle = Rotation2d(0.1)
```

Comment:
would be better to just finish silently if nothing detected or go to a standard driving instead of "search"